include/device.h

```diff
@@ -229,6 +271,78 @@ struct device {
 #endif
 };
 
+/**
+ * @brief Get the handle for a given device
+ *
+ * @param dev the device for which a handle is desired.
+ *
+ * @return the handle for the device.
+ */
+static inline device_handle_t
+device_get_handle(const struct device *dev)
+{
+	extern const struct device __device_start[];
+
+	/* TODO: If/when devices can be constructed that are not part of the
+	 * fixed we'll need another solution.
+	 */
+	return (device_handle_t)(dev - __device_start);
+}
+
+/**
+ * @brief Get the device corresponding to a handle.
+ *
+ * @param dh the device handle
+ *
+ * @return the device that has that handle, or a null pointer if @p dh does not
+ * identify a device.
+ */
+static inline const struct device *
+device_from_handle(device_handle_t dh)
+{
+	extern const struct device __device_start[];
+	extern const struct device __device_end[];
+	const struct device *dev = NULL;
+	size_t numdev = __device_end - __device_start;
+
+	if (dh < numdev) {
+		dev = &__device_start[dh];
+	}
+
+	return dev;
+}
+
+/**
+ * @brief Get the set of devices on which a given device depends
+ *
+ * @param dev the device for which a dependencies are desired.
+ *
+ * @param len pointer to a place to store the number of devices provided at the
```

Comment:
s/len/count

Reply:
`len` seems to be used more than `count` in Zephyr API that stores array lengths, but if you really want this I'll change it.

Reply:
Then talk about array length in the documentation.

Reply:
I like the existing text, so it's `count` now, and we have a sequence of count handles.